### PR TITLE
Use retries for IMDS and STS creds providers.

### PR DESCRIPTION
**Pull request by JonathanHenson**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by singku-china** [comment]
Since we have a clear definition of what HTTP errors to retry for STS Web Identity Provider, shall we apply this retry strategy as well to it?

**Comment by JonathanHenson** [comment]
Yep, was just doing these two to provide a practical example for you.

File: source/credentials_provider_sts.c [hunk 1]
```diff
@@ -239,6 +253,36 @@ static bool s_on_node_encountered_fn(struct aws_xml_parser *parser, struct aws_x
     return true;
 }
 
+/* errors that mean something screwy was going on at the networking layer. */
+static inline bool s_is_transient_error(int error_code) {
+    return error_code == AWS_ERROR_HTTP_CONNECTION_CLOSED || error_code == AWS_ERROR_HTTP_STREAM_CLOSED ||
```
Comment (singku-china) [inline source/credentials_provider_sts.c:258]: `AWS_ERROR_HTTP_STREAM_CLOSED ` this error code is no longer existed. Not sure why the checks passed.

Comment (singku-china) [reply source/credentials_provider_sts.c:258]: Looks like this error code is specific to H2 and has been moved to h2 error code. Shall we remove it from here? Or shall we change it to AWS_ERROR_HTTP_SERVER_CLOSED, which can also be a networking layer error.

**Comment by singku-china** [comment]
Merged this PR to #72, Closing this

**Pull request closed without merging**